podaac/umms_updater/util/create_assoc.py

```diff
@@ -103,7 +105,8 @@ def sync_association(cmr_env, concept_id, current_token, association, timeout=30
                 LOGGER.info("Failed add association: concept_id being associated "
                             "may not be valid: %s", assoc_concept_id)
 
-        for assoc_concept_id in remove:
+        LOGGER.info("Allow association removal: %s", remove_association)
+        for assoc_concept_id in remove and remove_association:
```

Comment:
this isn't valid python to loop and if in one statement

Reply:
use an if statement and then move the line remove = list(set(current) - set(new)) into if statement, so we only get the remove list if we want to remove

Reply:
changed